plugins/common/shim/goshim.go

```diff
@@ -84,13 +84,13 @@ func (s *Shim) Run(pollInterval time.Duration) error {
 		if err != nil {
 			return fmt.Errorf("RunProcessor error: %w", err)
 		}
-	} else if s.Output != nil {
+	} else if s.Output != nil { //nolint: revive // "Simplifying" if c {...} else {... return } would not simplify anything at all in this case
 		err := s.RunOutput()
 		if err != nil {
 			return fmt.Errorf("RunOutput error: %w", err)
 		}
 	} else {
-		return fmt.Errorf("Nothing to run")
+		return fmt.Errorf("nothing to run")
 	}
```

Comment:
Not sure, you could do:
```golang
	} else if s.Output == nil {
		return fmt.Errorf("nothing to run")
	}
	err := s.RunOutput()
	if err != nil {
		return fmt.Errorf("RunOutput error: %w", err)
	}
```

Reply:
I was wondering for quite some time about this solution and I came to conclusion that is not as clear as original version :)

Reply:
I agree, but then maybe adapt the comment to explaining _why_ we are not simplifying. Something in the line of `// Not simplifying here to stay in the structure for better understanding the code` or similar.

Reply:
Great! Corrected.